### PR TITLE
Fix: When updating object images, old thumbnails are not deleted and updated

### LINKED ISSUE #31809: When updating object images, old thumbnails are not deleted and updated
### Steps to reproduce
1. Use 8.1 and enable new image system. Set up jpg and webp formats.
2. Visit manufacturer page in FO so that all thumbs are generated.
3. Check `/img/m/` folder and see that manufacturer X has images in all formats.
4. Now go to BO and change the image of this manufacturer.
5. Check `/img/m/` again and see that only JPG thumbnails were updated.

### Problem
- The problem is that we generate only JPG files in BO after uploading them, for performance purposes mostly.
- The situation is the same for manufacturers, categories, stores, suppliers - all objects where the ID of image stay the same.
- It's not a problem for product images because their ID changes and I think we generate all formats directly.

### Temporary solution
- Proper thumbnails are generated after manually regenerating them in `Image settings` in backoffice and clicking "Delete all previous thumbnails".

### Proper solution
- Delete all thumbnails of the given objects after uploading an image in backoffice.

Ping @matthieu-rolland @nicosomb
Originally reproduced by @web-burn in https://github.com/PrestaShop/PrestaShop/pull/31310#issuecomment-1468874335

### PrestaShop version(s) where the bug happened

8.1

### PR DESCRIPTION
**Pull request by Codencode**
Delete all thumbnails of the given objects after uploading an image in backoffice.

| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 8.1.x
| Type?             | bug fix
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes #31809
| Sponsor company   | Codencode snc

### CODE REVIEW

**Review by 0x346e3730 — DISMISSED** [review]
lgtm 👍

File: classes/ObjectModel.php [hunk 1]
```diff
@@ -1965,12 +1965,28 @@ public function deleteImage($force_delete = false)
                 return false;
             }
 
+            /** @var PrestaShop\PrestaShop\Core\Image\ImageFormatConfiguration $imageFormatConfiguration */
+            $imageFormatConfiguration = ServiceLocator::get('PrestaShop\PrestaShop\Core\Image\ImageFormatConfiguration');
+            $configuredImageFormats = $imageFormatConfiguration->getGenerationFormats();
```
Comment (kpodemski) [inline classes/ObjectModel.php]:
```suggestion
            $configuredImageFormats = $imageFormatConfiguration->getGenerationFormats();
```

Comment (Codencode) [reply classes/ObjectModel.php]: > 
Thanks @kpodemski!

**Review by kpodemski — DISMISSED** [review]

**Review by boherm — DISMISSED** [review]
LGTM, thanks @Codencode!

**Comment by Codencode** [comment]
Do I need to do anything else?
Thanks

**Comment by boherm** [comment]
Hi @Codencode!
No, we waiting for QA team to process tests on your PR.
When we have a GO, we will merge your work on 8.1.x branch!

**Comment by Codencode** [comment]
@boherm, ok thanks!

**Comment by hibatallahAouadni** [comment]
This PR will be tested when the issue is verified

**Comment by kpodemski** [comment]
@hibatallahAouadni why is it taking so long to validate this issue? 🤔

**Review by M0rgan01 — APPROVED** [review]

**Review by florine2623 — COMMENTED** [review]
Hello @Codencode ,

Thanks for the PR ^^

The auto tests are green : https://github.com/florine2623/testing_pr/actions/runs/5374939290

I still have the issue.
Here are the steps I followed :
1. Use 8.1 and enable new image system. Set up jpg and webp formats.
2. Visit manufacturer page in FO so that all thumbs are generated.
3. Check /img/m/ folder and see that manufacturer X has images in all formats. :x:
4. Go to Design > Image Settings > Regenerate thumbnails for Brands 
5. Check /img/m/ folder and see that manufacturer X has images in all formats. ✅ 
6. Now go to BO and change the image of this manufacturer.
7. Check /img/m/ again and see that only JPG thumbnails were updated. :x:
8. Go to Design > Image Settings > Regenerate thumbnails for Brands 
9. Check /img/m/ again and see that only JPG thumbnails were updated. :x: 

https://github.com/PrestaShop/PrestaShop/assets/16019289/c5ef4c63-3b2c-4735-bef6-11af806d1b31




Am I missing something ?
Thanks ^^

**Comment by Codencode** [comment]
> Hello @Codencode ,
> 
> Thanks for the PR ^^
> 
> The auto tests are green : https://github.com/florine2623/testing_pr/actions/runs/5374939290
> 
> I still have the issue. Here are the steps I followed :
> 
> 1. Use 8.1 and enable new image system. Set up jpg and webp formats.
> 2. Visit manufacturer page in FO so that all thumbs are generated.
> 3. Check /img/m/ folder and see that manufacturer X has images in all formats. ❌
> 4. Go to Design > Image Settings > Regenerate thumbnails for Brands
> 5. Check /img/m/ folder and see that manufacturer X has images in all formats. ✅
> 6. Now go to BO and change the image of this manufacturer.
> 7. Check /img/m/ again and see that only JPG thumbnails were updated. ❌
> 8. Go to Design > Image Settings > Regenerate thumbnails for Brands
> 9. Check /img/m/ again and see that only JPG thumbnails were updated. ❌
> 
>  Screen.Recording.2023-06-26.at.11.55.43.mov 
> Am I missing something ? Thanks ^^

Hello @florine2623,

I made the changes to make the changes work even in case of new image system.

Thank You.

**Review by kpodemski — APPROVED** [review]

**Review by matks — APPROVED** [review]
Thank you for the code update @Codencode

**Review by M0rgan01 — APPROVED** [review]

**Review by Hlavtox — CHANGES_REQUESTED** [review]
Hey @Codencode, one tiny issue I see. Are you sure that there will always be image formats configured?

I think that during generating, you should check if the feature is enabled and fallback to JPG, like we do in other places in the code.
```
$sfContainer = SymfonyContainer::getInstance();
if ($sfContainer->get('prestashop.core.admin.feature_flag.repository')->isEnabled(FeatureFlagSettings::FEATURE_FLAG_MULTIPLE_IMAGE_FORMAT)) {
  $configuredImageFormats = $sfContainer->get(ImageFormatConfiguration::class)->getGenerationFormats();
} else {
  $configuredImageFormats = ['jpg'];
}
```

And for deleting, we should delete `['jpg']` + `ImageFormatConfiguration::SUPPORTED_FORMATS`.

**Review by florine2623 — APPROVED** [review]
Hello @Codencode 

Link to auto tests OK : https://github.com/MhiriFaten/testing_pr/actions/runs/5519198537

I tested the PR, it is working as expected.
All thumbnails are regenerated with new image ✅ 

Waiting for your answer for @Hlavtox 's comment https://github.com/PrestaShop/PrestaShop/pull/32493#pullrequestreview-1524269051 before validating the PR ^^

**Comment by Codencode** [comment]
> configuredImageFormats

Hi @Hlavtox,
The configuration always returns at least the JPG type, in fact in the image configuration section it is not possible to remove the flag from the "Base JPEG/PNG" type.

I am attaching screenshots of the image configuration section.

Maybe I'm leaving something out?

Thank You.

Screenshot 2023-07-11 at 16-06-03 Image Settings • Codenshop

Screenshot details

**Review by Hlavtox — APPROVED** [review]

**Comment by Hlavtox** [comment]
@Codencode OK, the value SHOULD be there. :-)

**Pull request merged**